dbus_service.py

```diff
@@ -128,33 +146,60 @@ def __init__(
+
+    @staticmethod
+    def _handlechangedvalue(path, value):
+        logging.debug("someone else updated %s to %s", path, value)
+        return True  # accept the change
+
+    @staticmethod
+    def _get_config():
+        config = configparser.ConfigParser()
+        config.read(f"{(os.path.dirname(os.path.realpath(__file__)))}/config.ini")
+        return config
+
+    # read config file
     def _read_config_dtu(self, actual_inverter):
         config = self._get_config()
         self.pvinverternumber = actual_inverter
         self.dtuvariant = str(config["DEFAULT"]["DTU"])
         if self.dtuvariant not in (constants.DTUVARIANT_OPENDTU, constants.DTUVARIANT_AHOY):
-            raise ValueError(f"Error in config.ini: DTU must be one of {constants.DTUVARIANT_OPENDTU}, {constants.DTUVARIANT_AHOY}, {constants.DTUVARIANT_TEMPLATE}")
+            raise ValueError(
+                f"Error in config.ini: DTU must be one of \
+                {constants.DTUVARIANT_OPENDTU}, \
+                {constants.DTUVARIANT_AHOY}, \
+                {constants.DTUVARIANT_TEMPLATE}"
+            )
         self.deviceinstance = int(config[f"INVERTER{self.pvinverternumber}"]["DeviceInstance"])
-        self.acposition = int(get_config_value(config, "AcPosition", "INVERTER", self.pvinverternumber ))
+        self.acposition = int(get_config_value(config, "AcPosition", "INVERTER", self.pvinverternumber))
         self.signofliveinterval = config["DEFAULT"]["SignOfLifeLog"]
         self.useyieldday = int(config["DEFAULT"]["useYieldDay"])
         self.pvinverterphase = str(config[f"INVERTER{self.pvinverternumber}"]["Phase"])
-        self.host = get_config_value(config, "Host", "INVERTER", self.pvinverternumber )
-        self.username = get_config_value(config, "Username", "INVERTER", self.pvinverternumber )
-        self.password = get_config_value(config, "Password", "INVERTER", self.pvinverternumber )
-        self.digestauth = is_true(get_config_value(config, "DigestAuth", "INVERTER", self.pvinverternumber, False ))
+        self.host = get_config_value(config, "Host", "INVERTER", self.pvinverternumber)
+        self.username = get_config_value(config, "Username", "INVERTER", self.pvinverternumber)
+        self.password = get_config_value(config, "Password", "INVERTER", self.pvinverternumber)
+        self.digestauth = is_true(get_config_value(config, "DigestAuth", "INVERTER", self.pvinverternumber, False))
 
         try:
             self.max_age_ts = int(config["DEFAULT"]["MagAgeTsLastSuccess"])
-        except Exception:
+        except ValueError as ex:
+            logging.debug("MagAgeTsLastSuccess: %s", ex)
+            logging.debug("MagAgeTsLastSuccess not set, using default")
             self.max_age_ts = 600
 
-        try:
-            self.dry_run = "0" != config["DEFAULT"]["DryRun"]
-        except Exception:
-            self.dry_run = False
-
+        self.dry_run = is_true(config["DEFAULT"]["DryRun"])
```

Comment:
This way it crashes when someone has not set DryRun in the ini at all. I'd prefer being flexible here. Maybe some if in_array...

Reply:
Sorry, it's only now that I remember that we have get_config_value and def get_default_config (shame on me). The former should be the right one here

Reply:
I'm a little lost here,

Do you want me to undo the changes to:

```python
 try:
            self.dry_run = "0" != config["DEFAULT"]["DryRun"]
        except Exception:
            self.dry_run = False
```

Reply:
I've just pushed my proposal - but yet untested